configure.ac

```diff
@@ -276,8 +320,12 @@ fi
 #        Because libtool both intercepts -Wl and arbitrarily re-orders its
 #        command line inputs, we have to concoct a single expression to
 #        enforce linking that cannot be split, hence the below expression.
-LIBADD_SECP256K1="-Wl,secp256k1/src/libsecp256k1_la-secp256k1.${OBJEXT},secp256k1/src/libsecp256k1_precomputed_la-precomputed_ecmult_gen.${OBJEXT},secp256k1/src/libsecp256k1_precomputed_la-precomputed_ecmult.${OBJEXT}"
-AC_SUBST([LIBADD_SECP256K1])
+    libsecp256k1_LIBS="-Wl,secp256k1/src/libsecp256k1_la-secp256k1.${OBJEXT},secp256k1/src/libsecp256k1_precomputed_la-precomputed_ecmult_gen.${OBJEXT},secp256k1/src/libsecp256k1_precomputed_la-precomputed_ecmult.${OBJEXT}"
+    ac_configure_args="${ac_configure_args} --disable-shared --with-pic --with-bignum=no --enable-experimental --enable-module-ecdh --enable-module-recovery --enable-module-ecdsa-s2c --enable-module-rangeproof --enable-module-surjectionproof --enable-module-whitelist --enable-module-generator --enable-module-extrakeys --enable-module-schnorrsig ${secp256k1_test_opt} --enable-openssl-tests=no --enable-exhaustive-tests=no --enable-benchmark=no --disable-dependency-tracking ${secp_asm}"
```

Comment:
unrelated to this change, but `--enable-openssl-tests` is no longer supported in secp and should be removed (ideally in its own commit).

Setting these args before we initialize secp_asm/secp256k1_test_opt and export the various variables seems wrong, what am I missing?

Reply:
> `--enable-openssl-tests` is no longer supported in secp and should be removed (ideally in its own commit).

Indeed. Really in its own PR, as it's off-topic for this PR.

> Setting these args before we initialize secp_asm/secp256k1_test_opt and export the various variables seems wrong

You're right. This slipped past me. I'll move the `ac_configure_args` and `AC_CONFIG_SUBDIRS` to the bottom again, guarded by a conditional on `!LINK_SYSTEM_SECP256K1`.

Incidentally, abusing `ac_configure_args` in this way is problematic if the `Makefile` determines that it's out of date and needs to call `autoreconf` (et al.) to regenerate itself, as then it calls back to `configure` to reconfigure libwally-core, but at that point all these extraneous configure options, intended for libsecp256k1, are appended to the libwally-core configure line. Especially troublesome is tho `--disable-shared`, which causes the libwally-core build to produce a static library even when libwally-core was originally configured with `--enabled-shared`.

Have you considered using [`AX_SUBDIRS_CONFIGURE`](https://www.gnu.org/software/autoconf-archive/ax_subdirs_configure.html), which is designed to provide for exactly this use case of configuring an third-party sub-project with different configure options than the enclosing project? I almost went that route myself but decided to try to preserve what you already had in place so as to maximize the chances of getting my PR merged.

Reply:
AX_SUBDIRS_CONFIGURE looks like a cleaner way to handle secp configuring, I wasn't aware of it. I'm amenable to testing it if you'd like to go in that direction.